docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+
+The APIMan gateway is deployed in conjunction with a supported OpenShift router.  A typical request to a managed endpoint is initially routed to the gateway and then further routed by the gateway to the targeted service.  This flow can be seen in **Figure 2** where the APIMan gateway relies upon the service proxy to find pods.
+
+**Figure 2 - Managed Service Request Flow with APIMan Gateway and Router**
+
+![APIMan request flow with router](apiman_request_wi_router.png "Figure 2")
+
+
+### Service Annotations
+Services intended to be managed and exposed as API Endpoints will be annotated<sup>[5](#r5)</sup>.  The annotations are repeated below for convenience:
+
+```
+  apiVersion: "v1"
+  kind: "Service"
+  metadata: 
+    annotations: 
+      api.service.kubernetes.io/protocol: REST
+      api.service.kubernetes.io/scheme: http
+      api.service.kubernetes.io/path: cxfcdi
+      api.service.kubernetes.io/description-path: cxfcdi/swagger.json
+      api.service.kubernetes.io/description-language: SwaggerJSON
+```
+Additionally, services that should be managed by APIMan will be further annotated identifying who is managing the service.  The proposed annotation is:
+```
+  openshift.io/service-managed-by: apiman
+```
+
+Service providers will need to explicitly publish a service using the APIMan user interface.  Future iterations may include functionality to automatically publish a service when these annotations are applied.
+
+### Origin Web Console UI extension
+A custom extension<sup>[6](#r6)</sup> will be created and hooked into the web console to support API endpoints.  The extension
+will allow a user to navigate to the APIMan management interface to:
+* Manage an API endpoint
+* Expose an annotated service as a new API endpoint
+
+The extension will provide the following details to the APIMan gateway:
+* Service name
+* Service namespace
+* User's Oauth token
+* Back link to the openshift web console
+
+Calls from the extension to the APIMan management interface will utilize a REST POST call where the oauth token is part of the payload.  It will
+utilize a similiar design<sup>[7](#r7)</sup> that is realized by the OpenShift origin aggregated logging integration and the auth proxy<sup>[8](#r8)</sup>.  It is necessary for
+security reasons to not add the token as a query parameter.  
+
+### API Services Gateway
+The APIMan gateway will be deployed as a cluster-wide infrastructure component to handle services exposed as API endpoints.  Internal and external clients can use the gateway to access APIs.  Admins can configure the cluster to limit direct access to services and encourage consumers of API endpoints to utilize the gateway by deploying the multi-tenant SDN plugin.  This plugin controls service to service communication between projects and would force traffic throught the gateway.  A route will be created as a well known endpoint to the APIMan gateway through which all API services traffic will flow.
```

Comment:
"encourage consumers of API endpoints to utilize the gateway" LOL more like "require"

also typo in "throught"